data-prepper-plugins/otel-trace-group-prepper/src/test/java/com/amazon/dataprepper/plugins/prepper/oteltracegroup/OtelTraceGroupPrepperTests.java

```diff
@@ -0,0 +1,256 @@
+
+    private OtelTraceGroupPrepper otelTraceGroupPrepper;
+    private ExecutorService executorService;
+
+    @Mock
+    private ConnectionConfiguration connectionConfigurationMock;
+
+    @Mock
+    private RestHighLevelClient restHighLevelClient;
+
+    @Mock
+    private SearchResponse testSearchResponse;
+
+    @Mock
+    private SearchHits testSearchHits;
+
+    @Mock
+    private SearchHit testSearchHit1;
+
+    @Mock
+    private SearchHit testSearchHit2;
+
+    @Before
+    public void setUp() throws Exception{
+        connectionConfigurationMockedStatic = Mockito.mockStatic(ConnectionConfiguration.class);
+        connectionConfigurationMockedStatic.when(() -> ConnectionConfiguration.readConnectionConfiguration(any(PluginSetting.class)))
+                .thenReturn(connectionConfigurationMock);
+        when(connectionConfigurationMock.createClient()).thenReturn(restHighLevelClient);
+        when(restHighLevelClient.search(any(SearchRequest.class), any(RequestOptions.class))).thenReturn(testSearchResponse);
+        doNothing().when(restHighLevelClient).close();
+        when(testSearchResponse.getHits()).thenReturn(testSearchHits);
+        when(testSearchHits.getHits()).thenReturn(new SearchHit[] {testSearchHit1});
+        when(testSearchHit1.field("traceId")).thenReturn(new DocumentField("traceId", Collections.singletonList(TEST_TRACE_ID_1)));
+        when(testSearchHit1.field("traceGroup")).thenReturn(new DocumentField("traceGroup", Collections.singletonList(TEST_TRACE_GROUP_1)));
+        when(testSearchHit2.field("traceId")).thenReturn(new DocumentField("traceId", Collections.singletonList(TEST_TRACE_ID_2)));
+        when(testSearchHit2.field("traceGroup")).thenReturn(new DocumentField("traceGroup", Collections.singletonList(TEST_TRACE_GROUP_2)));
+        final PluginSetting testPluginSetting = new PluginSetting("otel_trace_group_prepper", new HashMap<>()) {{
+            setPipelineName("testPipelineName");
+        }};
+        otelTraceGroupPrepper = new OtelTraceGroupPrepper(testPluginSetting);
+        executorService = Executors.newFixedThreadPool(TEST_NUM_WORKERS);
+    }
+
+    @After
+    public void tearDown() {
+        otelTraceGroupPrepper.shutdown();
+        connectionConfigurationMockedStatic.close();
+        executorService.shutdown();
+    }
+
+    @Test
+    public void testShutDown() throws IOException {
+        // When
+        otelTraceGroupPrepper.shutdown();
+
+        // Then
+        verify(restHighLevelClient, times(1)).close();
+    }
+
+    @Test
+    public void testTraceGroupFillSuccess() throws IOException {
+        // Given
+        Record<String> testRecord = buildRawSpanRecord(TEST_RAW_SPAN_MISSING_TRACE_GROUP_JSON_FILE_1);
+        List<Record<String>> testRecords = Collections.singletonList(testRecord);
+
+        // When
+        List<Record<String>> recordsOut = (List<Record<String>>) otelTraceGroupPrepper.doExecute(testRecords);
```

Comment:
Super nit - might leave out these comments altogether (or switch to something like Arrange/Act/Assert if you want to keep them). I'm mainly calling this out because when I read `// When` I immediately thought of the `when(mockedObject)` calls that happen as part of setting up a test. 

It was jarring to read `// When` but then see no `when()` calls after it.

Reply:
+1 for keeping the 3 sections of each test separate when possible, greatly improved readability